### PR TITLE
MAINT: upgrade netcdf4 requirement

### PR DESCRIPTION
**Pull request by Gui-FernandesBR**
## Pull request type

Please check the type of change your PR introduces:

- [x] Other (please describe): requirements update

## Pull request checklist

Please check if your PR fulfills the following requirements, depending on the type of PR:

- Code base maintenance (refactoring, formatting, renaming):

  - [x] Docs have been reviewed and added / updated if needed
  - [x] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [x] All tests (`pytest --runslow`) have passed locally

## What is the current behavior?
The requirements for netcdf4 dependency is a bit more complex than what would be ideal. This has been a problem since the BUG #205 was reported.

## What is the new behavior?
Everything is easier with the newest version of netcdf4, which may work without error all environments, including google colab.

## Does this introduce a breaking change?
- [x] No

## Other information
- It is sad that this PR isn't going to the master branch directly, but let's be patient and wait until the v1 gets finally merged into master
- This PR should wait for the #365 to be merged into beta/v1.0.0
- This should finally fix the BUG mentioned above (#205 )

### CODE REVIEW

File: docs/user/requirements.rst
Comment (MateusStano) [inline docs/user/requirements.rst:1]: Check out line 34 on this file. I think it also needs to be changed

**Review by MateusStano — CHANGES_REQUESTED** [review]
Just one change here, everything else seems good

File: docs/user/requirements.rst
Comment (Gui-FernandesBR) [reply docs/user/requirements.rst:1]: Nice catch, 
I just fixed it in the last commit.

Thanks for reviewing it =)

**Comment by Gui-FernandesBR** [comment]
Going to merge this one into `maint/drop-py37`, ad from there we merge into beta/v1. 
I think that's a faster way

**Pull request merged**